packages/fxa-shared/tracing/node-tracing.ts

```diff
@@ -128,9 +181,22 @@ export class NodeTracingInitializer {
     this.provider.register();
     this.logger?.info(log_type, { msg: 'gcp enabled' });
   }
+
+  ignoreTraceSample() {
```

Comment:
Instead of setting a unique attribute that we define, I'd recommend piggy packing off the `traceFlags` field. I think doing this would mean we wouldn't need the custom FxaSampler class. If you checkout this [test](https://github.com/open-telemetry/opentelemetry-js/blob/6111ae0181474f89c9f9441a0f8a474ef3092175/packages/opentelemetry-sdk-trace-base/test/common/sampler/ParentBasedSampler.test.ts#L53), it might give you some ideas. I found the `trace.setSpanContext(api.ROOT_CONTEXT, spanContext)` line particular interesting.



If this approach in fact works, I think it would be cleaner. Note that this [PR](https://github.com/mozilla/fxa/pull/14120), where we added a configurable sampling rate, introduced a `ParentBasedSampler`.